endpoints/openrtb2/auction.go

```diff
@@ -1391,22 +1394,47 @@ func (deps *endpointDeps) processStoredRequests(ctx context.Context, requestJson
 					err = fmt.Errorf("imp.ext.prebid.storedrequest.id %s: Stored Imp has Invalid JSON: %s", impIds[i], Err)
 				}
 			}
-			return nil, []error{err}
+			return nil, impToStoredReq, []error{err}
 		}
 		imps[idIndices[i]] = resolvedImp
+
+		includeStoredImps, err := jsonparser.GetBoolean(resolvedImp, "ext", "prebid", "options", "echovideoattrs")
+
+		if err != nil && err != jsonparser.KeyPathNotFoundError {
+			return nil, nil, []error{err}
+		}
+
+		//determine if this is video impression
+		_, _, offset, err := jsonparser.Get(resolvedImp, "video")
+		if err != nil && err != jsonparser.KeyPathNotFoundError {
+			return nil, nil, []error{err}
+		}
+
+		if includeStoredImps && offset != -1 {
+			impId, err := jsonparser.GetString(resolvedImp, "id")
+			if err != nil {
+				return nil, nil, []error{err}
+			}
+			//extract video attributes only
+			videoData, _, _, err := jsonparser.Get(storedImps[impIds[i]], "video")
+			if err == nil {
+				impToStoredReq[impId] = videoData
+			}
```

Comment:
To improve performance for non-video ads, please nest the other jsonparser checks under `includeStoredImps`. ie:

```
includeVideoAttributes, err := jsonparser.GetBoolean(resolvedImp, "ext", "prebid", "options", "echovideoattrs")
if err != nil && err != jsonparser.KeyPathNotFoundError {
	return nil, nil, []error{err}
}

if includeVideoAttributes {
	_, _, videoImpOffset, err := jsonparser.Get(resolvedImp, "video")
	if err != nil && err != jsonparser.KeyPathNotFoundError {
		return nil, nil, []error{err}
	}

	if videoImpOffset != -1 {
		impId, err := jsonparser.GetString(resolvedImp, "id")
		if err != nil {
			return nil, nil, []error{err}
		}
		//extract video attributes only
		videoData, _, _, err := jsonparser.Get(storedImps[impIds[i]], "video")
		if err == nil {
			impToStoredReq[impId] = videoData
		}

	}
}
```

Reply:
Yes, modified